multiqc/multiqc.py

```diff
@@ -362,33 +395,195 @@ def run(
+
+        _export_sources()
+
+        _write_json_dump()
+
+        _write_html_and_data(
+            tmp_dir=tmp_dir,
+            template_mod=template_mod,
+            filename=filename,
+        )
+
+    except _RunError as e:
+        if e.message:
+            logger.critical(e.message)
+        return RunResult(message=e.message, sys_exit_code=e.sys_exit_code)
+
+    plugin_hooks.mqc_trigger("execution_finish")
+
+    report.runtimes["total"] = time.time() - start_execution_time
+    if config.profile_runtime:
+        logger.warning(f"Run took {report.runtimes['total']:.2f} seconds")
+        logger.warning(f" - {report.runtimes['total_sp']:.2f}s: Searching files")
+        logger.warning(f" - {report.runtimes['total_mods']:.2f}s: Running modules")
+        if config.make_report:
+            logger.warning(f" - {report.runtimes['total_compression']:.2f}s: Compressing report data")
+            logger.info(f"For more information, see the 'Run Time' section in {os.path.relpath(config.output_fn)}")
+
+    if report.num_mpl_plots > 0 and not config.plots_force_flat:
+        if not config.plots_force_interactive:
+            console.print(
+                "[blue]|           multiqc[/] | "
+                "Flat-image plots used. Disable with '--interactive'. "
+                "See [link=https://multiqc.info/docs/#flat--interactive-plots]docs[/link]."
+            )
+
+    sys_exit_code = 0
+    if config.strict and len(report.lint_errors) > 0:
+        logger.error(f"Found {len(report.lint_errors)} linting errors!\n" + "\n".join(report.lint_errors))
+        sys_exit_code = 1
+
+    logger.info("MultiQC complete")
+
+    # Move the log file into the data directory
+    log.move_tmp_log(logger)
+
+    return RunResult(sys_exit_code=sys_exit_code)
+
+
+def _set_up_logging(verbose, quiet, no_ansi):
     # Set up logging level
     loglevel = log.LEVELS.get(min(verbose, 1), "INFO")
     if quiet:
         loglevel = "WARNING"
         config.quiet = True
     log.init_log(logger, loglevel=loglevel, no_ansi=no_ansi)
 
-    # Throw an error if we are using an unsupported version of Python
-    if sys.version_info < tuple(map(int, OLDEST_SUPPORTED_PYTHON_VERSION.split("."))):
-        logger.critical(
-            "You are running MultiQC with Python {}. "
-            "Please upgrade Python! MultiQC does not support Python < {}, "
-            "things will break.".format(sys.version_info, OLDEST_SUPPORTED_PYTHON_VERSION)
-        )
-        return {"report": None, "config": None, "sys_exit_code": 1}
-
     console = rich.console.Console(
         stderr=True,
         highlight=False,
         force_terminal=util_functions.force_term_colors(),
         color_system=None if no_ansi else "auto",
     )
-    console.print(
-        f"\n  [dark_orange]///[/] [bold][link=https://multiqc.info]MultiQC[/link][/] :mag: [dim]| v{config.version}\n"
-    )
-    logger.debug(f"This is MultiQC v{config.version}")
 
+    return console
+
+
+def _init(
```

Comment:
I find this a bit too close to the special `__init__`. Can we name it something different?

Reply:
This could probably also be moved, into the `config.py` file and renamed `init_config` or something, for example.